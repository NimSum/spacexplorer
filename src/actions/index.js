export const addUpcomingLaunches = launches => ({
  type: 'ADD_UPCOMING_LAUNCHES',
  launches
})

export const addSpaceEvents = events => ({
  type: 'ADD_SPACE_EVENTS',
  events
})

export const addSelectedLaunch = event => ({
  type: 'ADD_SELECTED_LAUNCH',
  event
})

export const toggleLaunchInfo = (bool) => ({
  type: 'TOGGLE_LAUNCH_INFO',
  bool: bool
})

export const addRockets = rockets => ({
  type: 'ADD_ROCKETS',
  rockets
})

export const addSpaceStations = spaceStations => ({
  type: 'ADD_SPACE_STATIONS',
  spaceStations
})

export const addOrbiters = orbiters => ({
  type: 'ADD_ORBITERS',
  orbiters
})

export const addAstronauts = astronauts => ({
  type: 'ADD_ASTRONAUTS',
  astronauts
})