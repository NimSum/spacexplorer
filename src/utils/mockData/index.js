export const mockAstronaut = {
    "id": 596,
    "url": "https://spacelaunchnow.me/api/3.3.1/astronaut/596/",
    "name": "Bill Nelson",
    "status": {
        "id": 2,
        "name": "Retired"
    },
    "type": {
        "id": 4,
        "name": "Non-Astronaut Passenger"
    },
    "agency": {
        "id": 44,
        "url": "https://spacelaunchnow.me/api/3.3.1/agencies/44/",
        "name": "National Aeronautics and Space Administration",
        "type": "Government"
    },
    "date_of_birth": "1942-09-29",
    "date_of_death": null,
    "nationality": "American",
    "twitter": null,
    "instagram": null,
    "bio": "Clarence William Nelson II (born September 29, 1942) is an American politician serving as the senior United States Senator from Florida, first elected to that seat in 2000. A member of the Democratic Party, he previously served in the Florida House of Representatives from 1972 to 1978 and in the United States House of Representatives from 1979 to 1991. In January 1986, he became the second sitting member of Congress to fly in space when he served as a payload specialist on the Space Shuttle Columbia.",
    "profile_image": "https://daszojo4xmsc6.cloudfront.net/media/astronaut_images/bill2520nelson_image_20190211162351.jpeg",
    "profile_image_thumbnail": "https://daszojo4xmsc6.cloudfront.net/media/default/cache/15/18/151821a1575b1c84201de6e3e4f11d73.jpg",
    "wiki": "https://en.wikipedia.org/wiki/Bill_Nelson",
    "flights": [
        {
            "id": "bbb4c25a-152d-4402-beee-2f62969b98c1",
            "url": "https://spacelaunchnow.me/api/3.3.1/launch/bbb4c25a-152d-4402-beee-2f62969b98c1/",
            "launch_library_id": 188,
            "slug": "https://spacelaunchnow.me/launch/space-shuttle-columbia-ov-102-sts-61-c-188",
            "name": "Space Shuttle Columbia / OV-102 | STS-61-C",
            "status": {
                "id": 3,
                "name": "Success"
            },
            "net": "1986-01-12T11:55:00Z",
            "window_end": "1986-01-12T11:55:00Z",
            "window_start": "1986-01-12T11:55:00Z",
            "mission": {
                "id": 682,
                "launch_library_id": 920,
                "name": "STS-61-C",
                "type": "Communications"
            },
            "pad": {
                "id": 87,
                "name": "Launch Complex 39A, Kennedy Space Center, FL",
                "location": {
                    "id": 27,
                    "name": "Kennedy Space Center, FL, USA"
                }
            },
            "orbit": "LEO",
            "rocket": {
                "id": 314,
                "configuration": {
                    "id": 170,
                    "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/170/",
                    "name": "Space Shuttle Columbia OV-102",
                    "full_name": "Space Shuttle Columbia OV-102",
                    "launch_service_provider": {
                        "id": 44,
                        "url": "https://spacelaunchnow.me/api/3.3.1/agencies/44/",
                        "name": "National Aeronautics and Space Administration",
                        "type": "Government"
                    },
                    "image_url": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/space2520shuttle2520columbia2520ov-102_image_20190222030326.jpeg"
                }
            }
        }
    ],
    "landings": [
        {
            "id": 67,
            "url": "https://spacelaunchnow.me/api/3.3.1/spacecraft/flight/67/",
            "destination": "Low Earth Orbit",
            "mission_end": "1986-01-18T13:58:00Z",
            "spacecraft": {
                "id": 36,
                "url": "https://spacelaunchnow.me/api/3.3.1/spacecraft/36/",
                "name": "Space Shuttle Columbia",
                "serial_number": "OV-102",
                "status": {
                    "id": 3,
                    "name": "Destroyed"
                },
                "description": "Space Shuttle Columbia (Orbiter Vehicle Designation: OV-102) was the first space-rated orbiter in NASA's Space Shuttle fleet. It launched for the first time on mission STS-1 on April 12, 1981, the first flight of the Space Shuttle program. Over 22 years of service, it completed 27 missions before disintegrating during re-entry near the end of its 28th mission, STS-107 on February 1, 2003, resulting in the deaths of all seven crew members.",
                "configuration": {
                    "id": 14,
                    "url": "https://spacelaunchnow.me/api/3.3.1/config/spacecraft/14/",
                    "name": "Space Shuttle",
                    "type": {
                        "id": 1,
                        "name": "Unknown"
                    },
                    "agency": {
                        "id": 44,
                        "url": "https://spacelaunchnow.me/api/3.3.1/agencies/44/",
                        "name": "National Aeronautics and Space Administration",
                        "type": "Government"
                    },
                    "in_use": false,
                    "image_url": "https://daszojo4xmsc6.cloudfront.net/media/orbiter_images/space2520shuttle_image_20190207032524.jpeg"
                }
            },
            "launch": {
                "id": "bbb4c25a-152d-4402-beee-2f62969b98c1",
                "url": "https://spacelaunchnow.me/api/3.3.1/launch/bbb4c25a-152d-4402-beee-2f62969b98c1/",
                "launch_library_id": 188,
                "slug": "https://spacelaunchnow.me/launch/space-shuttle-columbia-ov-102-sts-61-c-188",
                "name": "Space Shuttle Columbia / OV-102 | STS-61-C",
                "status": {
                    "id": 3,
                    "name": "Success"
                },
                "net": "1986-01-12T11:55:00Z",
                "window_end": "1986-01-12T11:55:00Z",
                "window_start": "1986-01-12T11:55:00Z",
                "mission": "STS-61-C",
                "mission_type": "Communications",
                "pad": "Launch Complex 39A, Kennedy Space Center, FL",
                "location": "Kennedy Space Center, FL, USA",
                "landing": null,
                "landing_success": null,
                "launcher": null,
                "orbit": "LEO",
                "image": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/space2520shuttle2520columbia2520ov-102_image_20190222030326.jpeg"
            }
        }
    ]
  }

export const mockEvents = {
  "count": 4,
  "next": null,
  "previous": null,
  "results": [
      {
          "id": 18,
          "url": "https://spacelaunchnow.me/api/3.3.1/event/18/",
          "name": "SpaceX Cargo Dragon (CRS-17) Unberthing and Reentry",
          "type": {
              "id": 8,
              "name": "Undocking"
          },
          "description": "Filled with more than 4,200 pounds of valuable scientific experiments and other cargo, a SpaceX Dragon cargo spacecraft is set to return to Earth from the International Space Station Monday, June 3. NASA Television and the agency’s website will provide live coverage of the craft’s release scheduled for 12:05 p.m. EDT (16:05 p.m UTC).",
          "location": "International Space Station",
          "news_url": "https://www.nasa.gov/press-release/nasa-tv-set-to-air-us-cargo-ship-departure-from-space-station",
          "video_url": "https://www.youtube.com/watch?v=21X5lGlDOfg",
          "feature_image": "https://daszojo4xmsc6.cloudfront.net/media/event_images/spacex2520cargo2520dragon25202528crs-172529_image_20190530165550.jpeg",
          "date": "2019-06-03T15:45:00Z",
          "launches": [
              {
                  "id": "de607493-4c2d-458e-9797-097732027ece",
                  "url": "https://spacelaunchnow.me/api/3.3.1/launch/de607493-4c2d-458e-9797-097732027ece/",
                  "launch_library_id": 1501,
                  "slug": "https://spacelaunchnow.me/launch/falcon-9-block-5-spx-crs-17-1501",
                  "name": "Falcon 9 Block 5 | SpX CRS-17",
                  "status": {
                      "id": 3,
                      "name": "Success"
                  },
                  "net": "2019-05-04T06:48:58Z",
                  "window_end": "2019-05-04T06:48:58Z",
                  "window_start": "2019-05-04T06:48:58Z",
                  "mission": "SpX CRS-17",
                  "mission_type": "Resupply",
                  "pad": "Space Launch Complex 40, Cape Canaveral, FL",
                  "location": "Cape Canaveral, FL, USA",
                  "landing": "OCISLY",
                  "landing_success": 1,
                  "launcher": "B1056",
                  "orbit": "LEO",
                  "image": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/falcon25209_image_20190224025007.jpeg"
              }
          ],
          "expeditions": [],
          "spacestations": [
              {
                  "id": 4,
                  "url": "https://spacelaunchnow.me/api/3.3.1/spacestation/4/",
                  "name": "International Space Station",
                  "status": {
                      "id": 1,
                      "name": "Active"
                  },
                  "founded": "1998-11-20",
                  "description": "The International Space Station (ISS) is a space station, or a habitable artificial satellite, in low Earth orbit. Its first component was launched into orbit in 1998, with the first long-term residents arriving in November 2000. It has been inhabited continuously since that date. The last pressurised module was fitted in 2011, and an experimental inflatable space habitat was added in 2016. The station is expected to operate until 2030. Development and assembly of the station continues, with several new elements scheduled for launch in 2019. The ISS is the largest human-made body in low Earth orbit and can often be seen with the naked eye from Earth. The ISS consists of pressurised habitation modules, structural trusses, solar arrays, radiators, docking ports, experiment bays and robotic arms. ISS components have been launched by Russian Proton and Soyuz rockets, and American Space Shuttles.",
                  "orbit": "Low Earth Orbit",
                  "image_url": "https://daszojo4xmsc6.cloudfront.net/media/spacestation_images/international2520space2520station_image_20190220215716.jpeg"
              }
          ]
      },
      {
          "id": 16,
          "url": "https://spacelaunchnow.me/api/3.3.1/event/16/",
          "name": "Progress MS-10 Undocking",
          "type": {
              "id": 8,
              "name": "Undocking"
          },
          "description": "The Progress MS-10 spacecraft will undock from the 'Zvezda aft' port on the International Space Station.\r\n\r\nIt will then de-orbit and burn up in the atmosphere.\r\n\r\n(current date is NET)",
          "location": "International Space Station",
          "news_url": "https://en.wikipedia.org/wiki/Progress_MS-10",
          "video_url": "https://www.youtube.com/watch?v=21X5lGlDOfg",
          "feature_image": "https://daszojo4xmsc6.cloudfront.net/media/event_images/progress2520ms-102520undocking_image_20190506111616.jpeg",
          "date": "2019-06-04T00:00:00Z",
          "launches": [
              {
                  "id": "d9ee0da9-b380-4e50-9f50-2a62a8e740e3",
                  "url": "https://spacelaunchnow.me/api/3.3.1/launch/d9ee0da9-b380-4e50-9f50-2a62a8e740e3/",
                  "launch_library_id": 1592,
                  "slug": "https://spacelaunchnow.me/launch/soyuz-fg-progress-ms-10-71p-1592",
                  "name": "Soyuz-FG | Progress MS-10 (71P)",
                  "status": {
                      "id": 3,
                      "name": "Success"
                  },
                  "net": "2018-11-16T18:14:08Z",
                  "window_end": "2018-11-16T18:14:08Z",
                  "window_start": "2018-11-16T18:14:08Z",
                  "mission": "Progress MS-10 (71P)",
                  "mission_type": "Resupply",
                  "pad": "1/5, Baikonur Cosmodrome, Kazakhstan",
                  "location": "Baikonur Cosmodrome, Republic of Kazakhstan",
                  "landing": null,
                  "landing_success": null,
                  "launcher": null,
                  "orbit": "LEO",
                  "image": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/soyuz_image_20190224025009.jpeg"
              }
          ],
          "expeditions": [],
          "spacestations": [
              {
                  "id": 4,
                  "url": "https://spacelaunchnow.me/api/3.3.1/spacestation/4/",
                  "name": "International Space Station",
                  "status": {
                      "id": 1,
                      "name": "Active"
                  },
                  "founded": "1998-11-20",
                  "description": "The International Space Station (ISS) is a space station, or a habitable artificial satellite, in low Earth orbit. Its first component was launched into orbit in 1998, with the first long-term residents arriving in November 2000. It has been inhabited continuously since that date. The last pressurised module was fitted in 2011, and an experimental inflatable space habitat was added in 2016. The station is expected to operate until 2030. Development and assembly of the station continues, with several new elements scheduled for launch in 2019. The ISS is the largest human-made body in low Earth orbit and can often be seen with the naked eye from Earth. The ISS consists of pressurised habitation modules, structural trusses, solar arrays, radiators, docking ports, experiment bays and robotic arms. ISS components have been launched by Russian Proton and Soyuz rockets, and American Space Shuttles.",
                  "orbit": "Low Earth Orbit",
                  "image_url": "https://daszojo4xmsc6.cloudfront.net/media/spacestation_images/international2520space2520station_image_20190220215716.jpeg"
              }
          ]
      }
  ]
}

export const mockLaunches = {
  "count": 204,
  "next": "https://spacelaunchnow.me/api/3.3.1/launch/upcoming/?limit=20&mode=detailed&offset=20",
  "previous": null,
  "results": [
      {
          "id": "41077f29-434a-4e53-9360-ca754f37e052",
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/41077f29-434a-4e53-9360-ca754f37e052/",
          "launch_library_id": 1927,
          "slug": "https://spacelaunchnow.me/launch/long-march-11-jilin-1-1927",
          "name": "Long March 11 | Jilin-1",
          "img_url": null,
          "status": {
              "id": 1,
              "name": "Go"
          },
          "net": "2019-06-05T04:00:00Z",
          "window_end": "2019-06-05T04:00:00Z",
          "window_start": "2019-06-05T04:00:00Z",
          "inhold": false,
          "tbdtime": false,
          "tbddate": false,
          "probability": -1,
          "holdreason": null,
          "failreason": null,
          "hashtag": null,
          "rocket": {
              "configuration": {
                  "id": 47,
                  "launch_library_id": 71,
                  "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/47/",
                  "name": "Long March 11",
                  "description": "",
                  "family": "Long March",
                  "full_name": "Long March 11",
                  "launch_service_provider": {
                      "id": 17,
                      "url": "https://spacelaunchnow.me/api/3.3.1/agencies/17/",
                      "name": "China National Space Administration",
                      "featured": true,
                      "type": "Government",
                      "country_code": "CHN",
                      "abbrev": "CNSA",
                      "description": "The China National Space Administration (CNSA) is the national space agency of the People's Republic of China. It is responsible for the national space program and for planning and development of space activities. CNSA and China Aerospace Corporation (CASC) assumed the authority over space development efforts previously held by the Ministry of Aerospace Industry. The CNSA has many launch sites around China with their orbital sites located in Jiuquan, Xichang, Taiyuan and Wenchang.",
                      "administrator": "Administrator: Tang Dengjie",
                      "founding_year": "1993",
                      "launchers": "Long March",
                      "spacecraft": "Shenzhou",
                      "launch_library_url": "https://launchlibrary.net/1.4/agency/17",
                      "successful_launches": 62,
                      "failed_launches": 1,
                      "pending_launches": 2,
                      "info_url": "http://www.cnsa.gov.cn/",
                      "wiki_url": "http://en.wikipedia.org/wiki/China_National_Space_Administration",
                      "logo_url": "https://daszojo4xmsc6.cloudfront.net/media/logo/china2520national2520space2520administration_logo_20190207032431.png",
                      "image_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_images/china2520national2520space2520administration_image_20190207032431.jpeg",
                      "nation_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_nation/china2520national2520space2520administration_nation_20190602114400.png"
                  },
                  "variant": "11",
                  "alias": "",
                  "min_stage": 1,
                  "max_stage": 4,
                  "length": 20,
                  "diameter": 2,
                  "maiden_flight": null,
                  "launch_mass": 58,
                  "leo_capacity": 500,
                  "gto_capacity": 0,
                  "to_thrust": 1177,
                  "apogee": 200,
                  "vehicle_range": 200,
                  "image_url": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/long2520march252011_image_20190222031217.jpeg",
                  "info_url": null,
                  "wiki_url": "https://en.wikipedia.org/wiki/Long_March_11"
              },
              "launcher_stage": [],
              "spacecraft_stage": null
          },
          "mission": {
              "id": 990,
              "launch_library_id": 1200,
              "name": "Jilin-1",
              "description": "The first sea launch of Long March 11 rocket. It carries two Jilin-1 Earth observation satellites.",
              "type": "Earth Science",
              "orbit": "Sun-Synchronous Orbit",
              "orbit_abbrev": "SSO"
          },
          "pad": {
              "id": 101,
              "agency_id": null,
              "name": "Unknown Pad",
              "info_url": null,
              "wiki_url": "",
              "map_url": "",
              "latitude": "0",
              "longitude": "0",
              "location": {
                  "id": 3,
                  "name": "Sea Launch",
                  "country_code": "UNK"
              }
          },
          "infoURLs": [],
          "vidURLs": []
      },
      {
          "id": "7bc965d8-9821-472e-aeba-f521ae2c06cf",
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/7bc965d8-9821-472e-aeba-f521ae2c06cf/",
          "launch_library_id": 1931,
          "slug": "https://spacelaunchnow.me/launch/hyperbola-1-7-unknown-payloads-1931",
          "name": "Hyperbola-1 | 7 unknown payloads",
          "img_url": null,
          "status": {
              "id": 2,
              "name": "TBD"
          },
          "net": "2019-06-10T02:00:00Z",
          "window_end": "2019-06-10T02:00:00Z",
          "window_start": "2019-06-10T02:00:00Z",
          "inhold": false,
          "tbdtime": true,
          "tbddate": true,
          "probability": -1,
          "holdreason": null,
          "failreason": null,
          "hashtag": null,
          "rocket": {
              "configuration": {
                  "id": 193,
                  "launch_library_id": 222,
                  "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/193/",
                  "name": "Hyperbola-1",
                  "description": "i-Space's first orbital rocket.",
                  "family": "Hyperbola",
                  "full_name": "Hyperbola-1",
                  "launch_service_provider": {
                      "id": 274,
                      "url": "https://spacelaunchnow.me/api/3.3.1/agencies/274/",
                      "name": "iSpace",
                      "featured": false,
                      "type": null,
                      "country_code": "CHN",
                      "abbrev": "iSpace",
                      "description": "i-Space is a Chinese private space launch company based in Beijing and founded in October 2016.",
                      "administrator": null,
                      "founding_year": "2016",
                      "launchers": "Hyperbola-1",
                      "spacecraft": "",
                      "launch_library_url": "https://launchlibrary.net/1.4/agency/274",
                      "successful_launches": 0,
                      "failed_launches": 0,
                      "pending_launches": 1,
                      "info_url": null,
                      "wiki_url": "https://en.wikipedia.org/wiki/I-Space_(Chinese_company)",
                      "logo_url": null,
                      "image_url": null,
                      "nation_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_nation/ispace_nation_20190602114618.png"
                  },
                  "variant": "1",
                  "alias": "",
                  "min_stage": null,
                  "max_stage": null,
                  "length": null,
                  "diameter": null,
                  "maiden_flight": null,
                  "launch_mass": null,
                  "leo_capacity": 300,
                  "gto_capacity": null,
                  "to_thrust": null,
                  "apogee": null,
                  "vehicle_range": null,
                  "image_url": null,
                  "info_url": null,
                  "wiki_url": null
              },
              "launcher_stage": [],
              "spacecraft_stage": null
          },
          "mission": null,
          "pad": {
              "id": 71,
              "agency_id": null,
              "name": "Unknown Pad, Jiuquan",
              "info_url": null,
              "wiki_url": "",
              "map_url": "",
              "latitude": "40.958",
              "longitude": "100.291",
              "location": {
                  "id": 17,
                  "name": "Jiuquan, People's Republic of China",
                  "country_code": "CHN"
              }
          },
          "infoURLs": [],
          "vidURLs": []
      },
      {
          "id": "6619844e-4777-4f7f-81d6-9ef0c3308d1d",
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/6619844e-4777-4f7f-81d6-9ef0c3308d1d/",
          "launch_library_id": 1937,
          "slug": "https://spacelaunchnow.me/launch/kuaizhou-11-maiden-flight-1937",
          "name": "Kuaizhou-11 | Maiden Flight",
          "img_url": null,
          "status": {
              "id": 2,
              "name": "TBD"
          },
          "net": "2019-06-15T00:00:00Z",
          "window_end": "2019-06-15T00:00:00Z",
          "window_start": "2019-06-15T00:00:00Z",
          "inhold": false,
          "tbdtime": true,
          "tbddate": true,
          "probability": -1,
          "holdreason": null,
          "failreason": null,
          "hashtag": null,
          "rocket": {
              "configuration": {
                  "id": 194,
                  "launch_library_id": 223,
                  "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/194/",
                  "name": "Kuaizhou-11",
                  "description": "",
                  "family": "",
                  "full_name": "",
                  "launch_service_provider": {
                      "id": 194,
                      "url": "https://spacelaunchnow.me/api/3.3.1/agencies/194/",
                      "name": "ExPace",
                      "featured": false,
                      "type": "Commercial",
                      "country_code": "CHN",
                      "abbrev": "EP",
                      "description": null,
                      "administrator": null,
                      "founding_year": null,
                      "launchers": "",
                      "spacecraft": "",
                      "launch_library_url": "https://launchlibrary.net/1.4/agency/194",
                      "successful_launches": 0,
                      "failed_launches": 0,
                      "pending_launches": 1,
                      "info_url": "",
                      "wiki_url": "https://en.wikipedia.org/wiki/ExPace",
                      "logo_url": null,
                      "image_url": null,
                      "nation_url": null
                  },
                  "variant": "",
                  "alias": "",
                  "min_stage": null,
                  "max_stage": null,
                  "length": null,
                  "diameter": null,
                  "maiden_flight": null,
                  "launch_mass": null,
                  "leo_capacity": null,
                  "gto_capacity": null,
                  "to_thrust": null,
                  "apogee": null,
                  "vehicle_range": null,
                  "image_url": null,
                  "info_url": "",
                  "wiki_url": ""
              },
              "launcher_stage": [],
              "spacecraft_stage": null
          },
          "mission": {
              "id": 1003,
              "launch_library_id": 1212,
              "name": "Maiden Flight",
              "description": "First flight of the new solid launcher developed by ExPace, subsidiary of CASIC. It will carry 2 communication satellites on this launch.",
              "type": "Test Flight",
              "orbit": "Low Earth Orbit",
              "orbit_abbrev": "LEO"
          },
          "pad": {
              "id": 71,
              "agency_id": null,
              "name": "Unknown Pad, Jiuquan",
              "info_url": null,
              "wiki_url": "",
              "map_url": "",
              "latitude": "40.958",
              "longitude": "100.291",
              "location": {
                  "id": 17,
                  "name": "Jiuquan, People's Republic of China",
                  "country_code": "CHN"
              }
          },
          "infoURLs": [],
          "vidURLs": []
      }
  ]
}