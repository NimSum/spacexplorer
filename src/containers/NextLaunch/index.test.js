import React from 'react';
import { toggleLaunchInfo } from '../../actions'
import { shallow } from 'enzyme';
import { NextLaunch, mapDispatchToProps, mapStateToProps } from './index';
import { mockLaunches } from '../../utils/mockData';


describe('NextLaunch', () => {
  let mockRocketLaunch = mockLaunches.results[0];
  let mockToggleLaunchInfo = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      < NextLaunch
        rocketLaunch={ mockRocketLaunch }
        showInfo={ false }
        toggleLaunchInfo={ mockToggleLaunchInfo }
      />
    )
  })
  
  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot with showInfo set to true', () => {
    wrapper.setProps({ showInfo: true });
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot if there is no rocketLaunch prop', () => {
    wrapper.setProps({ rocketLaunch: {} });
    expect(wrapper).toMatchSnapshot();
  })

})