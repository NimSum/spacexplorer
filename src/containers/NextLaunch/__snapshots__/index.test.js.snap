// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextLaunch should match component snapshot 1`] = `
<section
  className="next-launch-container"
>
  <article
    className="upcoming-launch-card"
  >
    <h3>
      Upcoming Rocket Launch:
    </h3>
    <h2>
      Long March 11 | Jilin-1
    </h2>
    <p>
      <span>
        Status: 
      </span>
       
      Go
    </p>
    <p>
      <span>
        Mission Type: 
      </span>
       
      Earth Science
    </p>
    <p>
      <span>
        Location: 
      </span>
       
      Sea Launch
    </p>
    <button
      onClick={[Function]}
    >
      More
    </button>
  </article>
  <CountDownTimer
    date="2019-06-05T04:00:00Z"
  />
  <img
    alt="rocket"
    src="https://daszojo4xmsc6.cloudfront.net/media/launcher_images/long2520march252011_image_20190222031217.jpeg"
  />
</section>
`;

exports[`NextLaunch should match snapshot if there is no rocketLaunch prop 1`] = `
<div>
  LOOOOOOOODINGGGG
</div>
`;

exports[`NextLaunch should match snapshot with showInfo set to true 1`] = `
<section
  className="next-launch-container"
>
  <Connect(NextLaunchInfo) />
  <CountDownTimer
    date="2019-06-05T04:00:00Z"
  />
  <img
    alt="rocket"
    src="https://daszojo4xmsc6.cloudfront.net/media/launcher_images/long2520march252011_image_20190222031217.jpeg"
  />
</section>
`;
